Soulstone V0.0.0.1

Taelina opened the pull request:
Alpha testing release, v0.0.0.1

Taelina committed:
Soulstone V0.0.0.1

reiichi001 commented:
Hi there! The `owners` field in your manifest.toml field is for github users. (You can add multiple github usernames there as an array value)

The build system won't engage when it's mismatched.

Taelina committed:
Merge branch 'goatcorp:main' into main

Taelina committed:
Fixing Manifest

Taelina committed:
Fixing manifest again.

Taelina committed:
Update to API 14

Taelina committed:
Merge branch 'main' of https://github.com/Taelina/DalamudPluginsD17

Taelina commented:
> Hi, currently reviewing this and noticed that the current code is not updated for API14, if you could do that I can then review the diff of the new commit once done.
> 
> Thanks!

Hi, hum, it should be the updated version, but I might have missed something? How do I make sure it's updated to API14?

Taelina commented:
I've already updated this PR, and the commit should already point to the commit where I updated to 14.0 API, so... Am I missing something ? Or have I done something wrong ? (And the plugin has been tested on my machine)

reiichi001 commented:
@Taelina, I think I see where the mixup is. Your `master` branch is still on API 13, which is where the `repo` link will head, despite the commit you've submitted being on your `taelina-dev` branch.

The plugin build system is inherently tied to the commit hash, as we must be able to certify that the binaries built match the source code commit from these PRs. 

As far as the review process goes, testing must be done by a third party to the developer[s] of the plugin. Usually, that's a review member, like myself, but occasionally we reach out to other plugin developers if the need arises.

Taelina committed:
Update to API 14 on Main branch

reiichi001 commented:
Hi there! Apologies on how long it's taken to get this plugin through review. We've been working on getting some more members onboarded for both PAC and a PAC-CodeReview team, to help speed this process up.

If you're still interested in getting this plugin on the main repository, we'd like to see Infi's comments addressed. https://github.com/goatcorp/DalamudPluginsD17/pull/7246#issuecomment-3885707384

We can then resume the review process from the last reviewed code commit, `2f039dd`.

Taelina commented:
It's fine, I'm also swamped and cannot work this much on this, but I'll eventually get to it.

As for a few of the comments I can immediately get to without getting to work : 
- ECommons is included for one (for now) instance of a tooltip. (Which will eventually get more widespread.)
- For the XIVChatEntry, I tried (around the time of the comment) making it a simple string, it didn't work sadly, so I'm at a loss
- For the loc stuff, I don't know Resx for now, but I might, in future versions look into it.
- The collection marshall is to get a ref to values inside a dictionnary, as I didn't find an other way to get those (but the values inside the dictionnary can't be null as I've ensured they never are.), however, if you know of a better method to get the ref of a value in a dictionnary, I'd gladly replace this.

Taelina committed:
Code Review changes

nathanctech commented:
Hello! There has been a new SDK update released for patch 7.5 which requires the SDK to be bumped to version `15.0.0`. Your plugin is still in the queue. Thanks!

Taelina committed:
Update to API 15 & Small fixes

reiichi001 commented:
bleatbot, rebuild

MidoriKami commented:
[Blocking] Remove the following line from your csproj, it is defined for you by the Dalamud.NET.Sdk:
```xml
  <ItemGroup>
    <PackageReference Update="DotNet.ReproducibleBuilds" Version="2.0.2" />
  </ItemGroup>
```

[Blocking] Remove ECommons from your project, you only use a single line of code from that library that you can easily replace with:
```cs
if (ImGui.IsItemHovered())
{
    ImGui.SetTooltip("La réputation de votre personnage dans son environnement social et professionnel.")
}
```

[Non Blocking] Remove `.github/workflows/pr_build.yml` or edit it to actually work for your project. It is targeting "SamplePlugin"

[Non Blocking] Remove `Data/goat.png` and the folder, as it's not used in your plugin at all.

[Non Blocking, Suggestion] Might consider wrapping any json serialization with try/catch to catch file exceptions without causing issues.

[Non Blocking, Recommendation] Recommend using ResX or CheapLoc for localization instead of a handwritten dictionary, this would also potentially allow others to contribute translations more easily, and can be integrated into Crowdin.

[Non Blocking] Remove `MaximumSize = new Vector2(float.MaxValue, float.MaxValue)` from `MainWindow.cs`, the default is unlimited max size.


Please correct the entries labeled [Blocking]
Optionally correct entries labeled [Non Blocking]
